domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterSingletonTest.kt

```diff
@@ -0,0 +1,140 @@
+const val stringPlatformParameterValue = "string_platform_parameter_value"
+
+const val integerPlatformParameterName = "integer_platform_parameter_name"
+const val integerPlatformParameterValue = 1
+
+const val booleanPlatformParameterName = "boolean_platform_parameter_name"
+const val booleanPlatformParameterValue = true
+
+const val incorrectPlatformParameterName = "incorrect_platform_parameter_name"
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PlatformParameterSingletonTest.TestApplication::class)
+class PlatformParameterSingletonTest {
+
+  @Inject
+  lateinit var platformParameterSingleton: PlatformParameterSingleton
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testSingleton_initialPlatformParameterMap_isEmpty(){
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isTrue()
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_isNotEmpty(){
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isTrue()
+    platformParameterSingleton.initPlatformParameterMap(makePlatformParameterMap())
+    assertThat(platformParameterSingleton.platformParameterMap.isEmpty()).isFalse()
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveStringParameter_verifyItsValue() {
+    platformParameterSingleton.initPlatformParameterMap(makePlatformParameterMap())
+    val stringPlatformParameter = platformParameterSingleton.getPlatformParameter<String>(
+      stringPlatformParameterName
+    )
+    assertThat(stringPlatformParameter?.value).isEqualTo(stringPlatformParameterValue)
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveIntegerParameter_verifyItsValue() {
+    platformParameterSingleton.initPlatformParameterMap(makePlatformParameterMap())
+    val integerPlatformParameter = platformParameterSingleton.getPlatformParameter<Int>(
+      integerPlatformParameterName
+    )
+    assertThat(integerPlatformParameter?.value).isEqualTo(integerPlatformParameterValue)
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveBooleanParameter_verifyItsValue() {
+    platformParameterSingleton.initPlatformParameterMap(makePlatformParameterMap())
+    val booleanPlatformParameter = platformParameterSingleton.getPlatformParameter<Boolean>(
+      booleanPlatformParameterName
+    )
+    assertThat(booleanPlatformParameter?.value).isEqualTo(booleanPlatformParameterValue)
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveIncorrectParameter_verifyItsValue() {
+    platformParameterSingleton.initPlatformParameterMap(makePlatformParameterMap())
+    val incorrectPlatformParameter = platformParameterSingleton.getPlatformParameter<String>(
+      incorrectPlatformParameterName
+    )
+    assertThat(incorrectPlatformParameter).isNull()
+  }
+
+  private fun makePlatformParameterMap(): Map<String, ParameterValue> {
```

Comment:
This might be better as a class constant since it's being used in nearly every test. That way we don't have to reconstruct it every time.

Reply:
Yeah, Done